com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/TextureGenerator.java

```diff
@@ -560,6 +477,65 @@ public static TextureImage generate(BufferedImage origImage, TextureProfile texP
         return generate(origImage, texProfile, compress, EnumSet.of(FlipAxis.FLIP_AXIS_Y));
     }
 
+    private static TextureImage.CompressionType textureFormatToSupportedCompressionTypeOrDefault(TextureImage.TextureFormat format, TextureImage.CompressionType defaultType) {
+        // Issue 5753: Since we currently don't support pre compressed hardware textures, so we use UASTC instead
+        if (format == TextureFormat.TEXTURE_FORMAT_RGB_PVRTC_2BPPV1 || format == TextureFormat.TEXTURE_FORMAT_RGB_PVRTC_4BPPV1 || format == TextureFormat.TEXTURE_FORMAT_RGB_ETC1) {
+            return TextureImage.CompressionType.COMPRESSION_TYPE_BASIS_UASTC;
+        } else if (format == TextureFormat.TEXTURE_FORMAT_RGBA_PVRTC_2BPPV1 || format == TextureFormat.TEXTURE_FORMAT_RGBA_PVRTC_4BPPV1) {
+            return TextureImage.CompressionType.COMPRESSION_TYPE_BASIS_UASTC;
+        }
+        return defaultType;
+    }
+
+    private static TextureImage.TextureFormat textureFormatToSupportedTextureFormat(TextureImage.TextureFormat format) {
+        // Issue 5753: Since we currently don't support pre compressed hardware textures, so we use UASTC instead
+        if (format == TextureFormat.TEXTURE_FORMAT_RGB_PVRTC_2BPPV1 || format == TextureFormat.TEXTURE_FORMAT_RGB_PVRTC_4BPPV1 || format == TextureFormat.TEXTURE_FORMAT_RGB_ETC1) {
+            return TextureFormat.TEXTURE_FORMAT_RGB;
+        } else if (format == TextureFormat.TEXTURE_FORMAT_RGBA_PVRTC_2BPPV1 || format == TextureFormat.TEXTURE_FORMAT_RGBA_PVRTC_4BPPV1) {
+            return TextureFormat.TEXTURE_FORMAT_RGBA;
+        }
+        return format;
+    }
+
+    private static String compressionTypeToTextureCompressor(TextureImage.CompressionType type) {
+        return switch (type) {
+            case COMPRESSION_TYPE_DEFAULT,
+                 COMPRESSION_TYPE_WEBP -> TextureCompressorDefault.TextureCompressorName;
```

Comment:
Should we use the ITextureCompressor.getName() api instead?

Reply:
We don't have a compressor instance available here, and we can't make it static due to other places where the instance getName() method is used. I think this is fine for now.